### PR TITLE
FloatMatrix * FloatMatrix multiplication optimization

### PR DESCRIPTION
**Pull request by davidhesselbom**
This greatly reduces the number of calls to `OwnedBuffer free`.

### CODE REVIEW

**Comment by davidhesselbom** [comment]
@tomasmahlberg Peer review?

File: source/math/FloatMatrix.ooc [hunk 1]
```diff
@@ -348,15 +348,16 @@ FloatMatrix : cover {
 				raise("Invalid dimensions in FloatMatrix * operator: left width must match right height!")
 		}
 		otherWidth := other take() width
+		(thisWidth, thisHeight) := (t width, t height)
 		result := This new(otherWidth, t height)
```
Comment (tomasmahlberg) [inline source/math/FloatMatrix.ooc]: `t height`=> `thisHeight` here also maybe.

Comment (davidhesselbom) [inline source/math/FloatMatrix.ooc]: Sure, I'll get that fixed, too.

**Comment by tomasmahlberg** [comment]
Did you try doing this for `upper` and `lower` `width` in `_forwardSubstitution` and `_backwardSubstitution` also? Could maybe save some additional calls to `free`.

**Comment by davidhesselbom** [comment]
I did try that, and it did remove even more `OwnedBuffer` calls. However, those two functions look very, very similar, and I think it would be better to somehow merge them before applying identical optimizations in both.

**Comment by tomasmahlberg** [comment]
Alright, this looks good now.

File: source/math/FloatMatrix.ooc [hunk 2]
```diff
@@ -250,7 +250,7 @@ FloatMatrix : cover {
 					accumulator -= lowerElements[x2 + y * lower take() width] * resultElements[x + x2 * result take() width]
 				value := lowerElements[y + y * lower take() width]
 				if (value != 0)
-					resultElements[x + y * result take()width] = accumulator / value
```
Comment (marcusnaslund) [inline source/math/FloatMatrix.ooc]: What? This works?

Comment (davidhesselbom) [inline source/math/FloatMatrix.ooc]: Sadly...

**Pull request merged**